hyperspy/drawing/figure.py

```diff
@@ -51,12 +76,18 @@ def _draw_animated(self):
                     [ax.draw_artist(a) for a in artists if
                      a.get_animated() is True]
                 except AttributeError:
-                     # The method was called before draw. This is a quick
-                     # fix. Properly fixing the issue involves avoiding
-                     # calling this method too early in the code.
+                    # The method was called before draw. This is a quick
+                    # fix. Properly fixing the issue involves avoiding
+                    # calling this method too early in the code.
+                    _logger.warning('Please report this warning to the '
+                                    'developpers: `_draw_animated` was called '
+                                    '`draw`.')
                     pass
             if canvas.supports_blit:
-                canvas.blit(self.figure.bbox)
+                if hasattr(canvas, 'update'):
+                    canvas.update()
```

Comment:
I cannot find any official documentation for this function. Could you link it? When will it be there, and what does it do differently than the blit?

Reply:
according to [this](https://github.com/matplotlib/matplotlib/pull/9505), is the `update` or `blit` even necessary? 

The relevant bit:
```
 +    In most backends callbacks subscribed to this callback will be
 +    fired after the rendering is complete but before the screen is
 +    updated.  Any extra artists drawn to the canvas's renderer will
 +    be reflected without an explicit call to ``blit``.
 +
 +    .. warning ::
 +
 +       Calling ``canvas.draw`` and ``canvas.blit`` in these callbacks may
 +       not be safe with all backends and may cause infinite recursion.
```

Reply:
This is the `update` method from QtWidget. For some reason, the `blit(self.figure.bbox)` is not working nicely anymore with the qt4 backend (I haven't tested qt5): it does not draw completely the plot but it does it after a draw event is triggered... the `update` method does the job as before though!

Reply:
As this has worked properly until recently, I that that you may have trying to workaround an external bug here. What is triggering the bug, the version of qt4 or matplotlib? Maybe the issue has been fixed externally in the mean time? In any case, would it be possible to simply skip the problematic version in ``setup.py``? If not,  I would suggest to  change the ``if hasattr(canvas, 'update')`` for checking if the backend is qt4 in order not to trigger the ``update`` unnecessarily for other backends.

Reply:
Yes, this was more or less a workaround and not a very good one, quite a few changes have been made for the qt backends in the last few matplotlib releases -not sure this is completely finished! I have not followed what's going on but there are some relevant information in matplotlib/matplotlib#9505 and matplotlib/matplotlib#9199.